### PR TITLE
expression: Add json_extract, cast(json as string), json_unquote push down support for tiflash

### LINKED ISSUE #39458: Support '->', '->>', 'json_extract' function push down to TiFlash
## Enhancement
https://dev.mysql.com/doc/refman/8.0/en/json-search-functions.html#function_json-extract
https://dev.mysql.com/doc/refman/8.0/en/json-search-functions.html#operator_json-inline-path

### PR DESCRIPTION
**Pull request by yibin87**
Signed-off-by: yibin <huyibin@pingcap.com>

### What problem does this PR solve?

Issue Number: close #39458  ref [](https://github.com/pingcap/tiflash/issues/6376)

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by LittleFall — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

File: expression/expr_to_pb_test.go [hunk 1]
```diff
@@ -958,6 +975,11 @@ func TestExprPushDownToFlash(t *testing.T) {
 
 	exprs = exprs[:0]
 
+	// json_unquote's argument is not cast(json as string)
+	function, err = NewFunction(mock.NewContext(), ast.JSONUnquote, types.NewFieldType(mysql.TypeString), stringColumn)
+	require.NoError(t, err)
+	exprs = append(exprs, function)
+
```
Comment (guo-shaoge) [inline expression/expr_to_pb_test.go:982]: Put these together with `json_unquote argument is cast(json as string)`?

**Review by guo-shaoge — APPROVED** [review]

**Review by mengxin9014 — APPROVED** [review]

**Pull request merged**